misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -56,6 +60,7 @@ public class JDIEventProcessor {
     private final BreakpointProcessor breakpointProcessor;
     private boolean isRemoteVmAttached = false;
     private final List<EventRequest> stepRequests = new ArrayList<>();
+    private static final List<ThreadReference> virtualThreads = new CopyOnWriteArrayList<>();
```

Comment:
Why CopyOnWriteArrayList?

Reply:
As the `virtual Threads` can be updated and accessed asynchronously, we need a thread-safe approach. 
I used `CopyOnWriteArrayList` because the `read` is more frequent than the `write` operations.